src/HPWHHeatSources.cc

```diff
@@ -11,8 +11,11 @@
 #include "HPWH.hh"
 
 // public HPWH::HeatSource functions
-HPWH::HeatSource::HeatSource(HPWH* parentInput)
-    : hpwh(parentInput)
+HPWH::HeatSource::HeatSource(
+    HPWH* parentInput,
+    const std::shared_ptr<Courier::Courier> courier /*std::make_shared<DefaultCourier>()*/)
+    : Sender("HeatSource", courier)
+    , hpwh(parentInput)
```

Comment:
The Sender class also has a `parent_pointer` member that allows the message context to be passed up through the hierarchy of classes. That should be set too.